tests/integration.rs

```diff
@@ -12,38 +12,59 @@ struct TestDefinition {
     query: String,
     input: String,
     output: String,
+    error: Option<String>,
     notes: Option<String>,
 }
 
 #[cfg(test)]
 mod integration {
     use super::*;
-    use ag::pipeline::Pipeline;
+    use ag::pipeline::{ErrorReporter, Pipeline, QueryContainer};
     use assert_cli;
     use std::borrow::Borrow;
     use toml;
 
+    pub struct EmptyErrorReporter;
+
+    impl ErrorReporter for EmptyErrorReporter {}
+
     fn structured_test(s: &str) {
         let conf: TestDefinition = toml::from_str(s).unwrap();
         let out: &str = conf.output.borrow();
+        let err = conf.error.unwrap_or("".to_string());
+        let env = assert_cli::Environment::inherit()
+            .insert("RUST_BACKTRACE", "0")
+            .insert("NO_COLOR", "");
         assert_cli::Assert::main_binary()
+            .with_env(&env)
             .stdin(conf.input)
             .with_args(&[&conf.query])
             .stdout()
             .is(out)
+            .stderr()
+            .is(err.as_str())
+            .ignore_status()
```

Comment:
Can you tweak this so that it only allows (and expects) an error status code if `Error` is defined on the structured test? I just want to make sure we have a test that verifies we have our status codes in line